lib/components/artist/shows/large_image_show.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+'use strict';
+
+import Relay from 'react-relay';
+import React from 'react-native';
+const { View, Text, StyleSheet, Dimensions } = React;
+
+import Headline from '../../text/headline'
+import SerifText from '../../text/serif'
+import OpaqueImageView from '../../opaque_image_view'
+import ShowMetadata from './metadata'
+
+const windowDimensions = Dimensions.get('window');
+const imageWidth = windowDimensions.width > 700 ? (windowDimensions.width - 85)/2 : (windowDimensions.width - 40);
```

Comment:
I would say, for now make that extra component. It’s fine to have some duplication for a while, then once you have everything implemented that you need, it will be much clearer what to abstract and how, as the overlap will be clear.